bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud/CloudUtil.java

```diff
@@ -47,24 +46,22 @@
 public class CloudUtil {
 
     private static final Random RANDOM = new Random();
+    private static final String DB_FOLDER_NAME = ConfigConstants.getUserDataFolder() + File.separator
+            + MiIoBindingConstants.BINDING_ID;
 
-    public static String getElementString(JsonElement jsonElement, String element, Logger logger) {
-        String value = "";
-        try {
-            value = jsonElement.getAsJsonObject().get(element).getAsString();
-        } catch (IllegalStateException | ClassCastException e) {
-            logger.debug("Json Element {} expected but missing", element);
-        }
-        return value;
-    }
-
-    public static void saveFile(String data, String country, Logger logger) {
-        String dbFolderName = ConfigConstants.getUserDataFolder() + File.separator + MiIoBindingConstants.BINDING_ID;
-        File folder = new File(dbFolderName);
+    /**
+     * Saves the Xiaomi cloud device info with tokens to file
+     *
+     * @param data file content
+     * @param country county server
+     * @param logger
+     */
+    public static void saveDeviceInfoFile(String data, String country, Logger logger) {
+        File folder = new File(DB_FOLDER_NAME);
         if (!folder.exists()) {
             folder.mkdirs();
         }
-        File dataFile = new File(dbFolderName + File.separator + "miioTokens-" + country + ".json");
+        File dataFile = new File(DB_FOLDER_NAME + File.separator + "miioTokens-" + country + ".json");
```

Comment:
You already have the parent file, so might as well use it.
```suggestion
        File dataFile = new File(folder, "miioTokens-" + country + ".json");
```

Reply:
darn... I should drink more coffee in the evening :-)